### PR TITLE
Split shplonk in prep for work queue

### PR DESCRIPTION
**Pull request by ledwards2225**
# Description

TLDR: Split Shplonk (just like we did for Gemini) to facilitate computation of commitments in work_queue.

The Shplonk prover has two main operations: 1) compute batched quotient polynomial Q, and 2) compute polynomial Q - Q_z. The challenge z at which Q is partially evaluated depends on the commitment [Q]. In order to compute this commitment in the work_queue, operations (2) must be separated from task (1). As part of this separation, all transcript operations have been moved to the `Prover` itself.

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [x] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [x] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

File: cpp/src/barretenberg/honk/proof_system/prover.hpp [hunk 1]
```diff
@@ -71,6 +73,9 @@ template <typename settings> class Prover {
     // Container for d + 1 Fold polynomials produced by Gemini
     std::vector<Polynomial> fold_polynomials;
 
+    Polynomial batched_quotient_Q;
```
Comment (ledwards2225) [inline cpp/src/barretenberg/honk/proof_system/prover.hpp:75]: I wonder if a more proper way to handle this would be to have a `std::shared_ptr<Polynomial>`, rather than moving it and leaving an empty poly behind. Not a huge deal but curious about best practice here. Thoughts @adr1anh?

Comment (adr1anh) [reply cpp/src/barretenberg/honk/proof_system/prover.hpp:75]: Maybe an optional here could more explicitly indicate whether the entry is used?

Comment (adr1anh) [reply cpp/src/barretenberg/honk/proof_system/prover.hpp:75]: Shared pointer wouldn't make much sense since it's not actually shared anywhere. It also has a bit of overhead from having to do reference counting.

Comment (adr1anh) [reply cpp/src/barretenberg/honk/proof_system/prover.hpp:75]: A unique pointer might make sense, but the main point it in checking that the element is null or not, in which case an optional is more explicit

**Comment by adr1anh** [comment]
LGTM

**Review by adr1anh — APPROVED** [review]

File: cpp/src/barretenberg/honk/proof_system/prover.hpp (hunk 1 above)
Comment (ledwards2225) [reply cpp/src/barretenberg/honk/proof_system/prover.hpp:75]: @adr1anh I'm sure I'm clear on what you mean -- I haven't seen `optional` applied to a class member before, only an input/output to a function. Is this an option or am I misunderstanding you? Either way, I'll leave it as is for now since it's hardly dangerous and I'm guessing this gets refactored soon anyway :)
